File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;
        for (int n : nums) {
            ++count[n];
        }
        
        vector<vector<int>> freq(nums.size() + 1);
        for (auto& [num, cnt]: count) {
            freq[cnt].push_back(num);
        }

        vector<int> result;
        for (int i = freq.size() - 1; i > 0; --i) {
            for (int num : freq[i]) {
                result.push_back(num);
                if (result.size() == k) {
                    return result;
                }
            }
        }
        return result; // should not reach here because of constraints, but added for completness
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-9.py
from collections import defaultdict

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        freq = defaultdict(int)

        for num in nums:
            freq[num] += 1

        for value in freq.values():
            if value > 1:
                return True

        return False
        

        
